### PR TITLE
Impute sex uses filter_intervals

### LINKED ISSUE #3380: impute_sex should use filter_intervals

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #3380

### CODE REVIEW

File: python/hail/methods/statgen.py [hunk 1]
```diff
@@ -201,30 +201,32 @@ def impute_sex(call, aaf_threshold=0.0, include_par=False, female_threshold=0.2,
...
+    mt, _ = mt._process_joins(call)
+    mt = mt.annotate_entries(call=call)
+    mt = require_biallelic(mt, 'impute_sex')
     if (aaf is None):
-        ds = ds.annotate_rows(aaf=agg.call_stats(ds.call, ds.alleles).AF[1])
+        mt = mt.annotate_rows(aaf=agg.call_stats(mt.call, mt.alleles).AF[1])
         aaf = 'aaf'
 
+    rg = mt.locus.dtype.reference_genome
     # FIXME: filter_intervals
```
Comment (catoverdrive) [inline python/hail/methods/statgen.py]: remove this `FIXME` now?

File: python/hail/methods/statgen.py [hunk 2]
```diff
@@ -201,30 +201,32 @@ def impute_sex(call, aaf_threshold=0.0, include_par=False, female_threshold=0.2,
...
-    ds = ds.filter_rows(ds[aaf] > aaf_threshold)
-    ds = ds.annotate_cols(ib=agg.inbreeding(ds.call, ds[aaf]))
-    kt = ds.select_cols(
-        *ds.col_key,
-        is_female=hl.cond(ds.ib.f_stat < female_threshold,
+    mt = hl.filter_intervals(mt,
+                             hl.map(lambda x_contig: hl.parse_locus_interval(x_contig, rg), rg.x_contigs),
+                             keep=True)
+    if not include_par:
+        mt = hl.filter_intervals(mt, rg.par, keep=False)
```
Comment (catoverdrive) [inline python/hail/methods/statgen.py]: rg.par looks like a list of tuple of `(contig, start, end)`, not a list of intervals (unless the documentation is wrong). Can you make sure this branch is tested in the impute_sex tests?

File: python/hail/methods/statgen.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: for some reason, my eyes just gloss over FIXMEs. This is the opposite of what is good.

File: python/hail/methods/statgen.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: the parameter `par` is a list of tuples, but the property `par` is a list of intervals. It is documented correctly, albeit confusingly.

Comment (tpoterba) [reply python/hail/methods/statgen.py]: confirmed that this branch is tested.

Comment (catoverdrive) [reply python/hail/methods/statgen.py]: oh, derp

**Pull request merged**